File: 2018/2/b.php
<?php
$input = [];
while($line = rtrim(fgets(STDIN))) {
    $input[] = $line;
}

for($i = 0 ; $i < count($input); $i++) {
    for($j = $i + 1 ; $j < count($input) ; $j++) {
        if(1 === levenshtein($input[$i], $input[$j])) {
            $first = $input[$i];
            $second = $input[$j];
            break 2;
        }
    }
}

$output = '';

for($i = 0 ; $i < strlen($first); $i++) {
    if($first[$i] === $second[$i]) {
        $output .= $first[$i];
    }
}

echo $output;

File: 2018/2/b.exs
input =
  File.read!("input")
  |> String.split(["\r", "\n"], trim: true)

defmodule B do
  def dist(first, second) do
    len = String.length(first) - 1

    for(x <- 0..len, String.at(first, x) != String.at(second, x), do: true)
    |> Enum.count()
  end

  def common(first, second) do
    len = String.length(first) - 1

    for(x <- 0..len, String.at(first, x) == String.at(second, x), do: String.at(first, x))
    |> Enum.join("")
  end
end

count = Enum.count(input) - 1

for(
  a <- 0..count,
  b <- 1..count,
  a < b,
  first = Enum.at(input, a),
  second = Enum.at(input, b),
  1 == B.dist(first, second),
  do: B.common(first, second)
)
|> Enum.at(0)
|> IO.inspect()
